### PR TITLE
Use pre-stored pointer to get arg and env

### PR DESCRIPTION
**Pull request by Marsman1996**
Use pre-stored pointer to get `arg` and `env` instead of fetching from stack, which prevents the corrupt by the user as 
https://github.com/asterinas/asterinas/pull/1567#issuecomment-2466244369

### CODE REVIEW

**Comment by tatetian** [comment]
I quote from https://github.com/asterinas/asterinas/pull/1567#issuecomment-2466244369:

> Directly reading the process stack without protection seems fragile and may introduce security risks. 

I don't think this is true. Even if you are not reading the `argc` from the stack, you still need to read the C strings from the stack. And in system call handling, we read all sorts of pointers from the user space and then use these user-space pointers to read data from the user space. Reading from the initial stack is not more risky than doing from user-given buffers as system call arguments.

In short, the changes proposed in this PR only provide a false sense of security or safety. Since you have mentioned that Linux does store `argv` and `envp` in kernel, I am still open to this idea. But I guess that there is probably a different reason. But without knowing a good reason, I don't feel comfortable adding the complexity and the memory footprint of `ArgEnvBoundaries`.

**Comment by Marsman1996** [comment]
Well, I will open another PR to fix the panic by adding more checks.
